### PR TITLE
Hosted youtube ophan tracking + refactor

### PR DESCRIPTION
**Pull request by Calanthe**
## What does this change?

Ophan tracking for hosted youtube videos. Refactor of the whole hosted video module included.
## What is the value of this and can you measure success?

Ophan tracking works on youtube videos

## Does this affect other platforms - Amp, Apps, etc?

no

## Screenshots

Example event:
screen shot 2016-10-03 at 17 39 20
## Request for comment

@lps88 @akash1810

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/hosted/next-video-autoplay.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    function cancelAutoplay() {
+        fastdom.write(function () {
+            $hostedNext.addClass('hosted-slide-out');
+        });
+        clearInterval(nextVideoInterval);
+    }
+
+    function cancelAutoplayMobile($hostedNext) {
+        fastdom.write(function () {
+            $hostedNext.addClass('u-h');
```
Comment (akash1810) [inline static/src/javascripts/projects/commercial/modules/hosted/next-video-autoplay.js]: can we use a more descriptive class name?

File: static/src/javascripts/projects/commercial/modules/hosted/next-video-autoplay.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            var timeLeft = duration - Math.floor(getCurrentTimeFn());
+            var countdownLength = 10; //seconds before the end when to show the timer
+
+            if (timeLeft <= countdownLength) {
+                fastdom.write(function () {
+                    $hostedNext.addClass('js-autoplay-start');
+                    $timer.text(timeLeft + 's');
+                });
+            }
+            if (timeLeft <= 0){
```
Comment (akash1810) [inline static/src/javascripts/projects/commercial/modules/hosted/next-video-autoplay.js]: would `timeLeft` ever be `<0`?

File: static/src/javascripts/projects/commercial/modules/hosted/youtube.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    function sendPercentageCompleteEvents(youtubePlayer, playerTotalTime) {
+        var quartile = playerTotalTime / 4;
+        var playbackEvents = {
+            'play': 0,
+            '25': quartile,
+            '50': quartile * 2,
+            '75': quartile * 3,
+            'end': playerTotalTime
```
Comment (akash1810) [inline static/src/javascripts/projects/commercial/modules/hosted/youtube.js]: does this still get fired when we autoplay the next video?

**Review by akash1810 — APPROVED** [review]
LGTM 👍

File: static/src/javascripts/projects/commercial/modules/hosted/next-video-autoplay.js (hunk 1 above)
Comment (Calanthe) [inline static/src/javascripts/projects/commercial/modules/hosted/next-video-autoplay.js]: This is a class commonly used in the whole frontend to hide elements.

Comment (akash1810) [inline static/src/javascripts/projects/commercial/modules/hosted/next-video-autoplay.js]: ah! my bad!

File: static/src/javascripts/projects/commercial/modules/hosted/next-video-autoplay.js (hunk 2 above)
Comment (Calanthe) [inline static/src/javascripts/projects/commercial/modules/hosted/next-video-autoplay.js]: There is `Math.floor` used a few lines above so probably not... But I just want to be extra safe ;)

File: static/src/javascripts/projects/commercial/modules/hosted/youtube.js (hunk 3 above)
Comment (Calanthe) [inline static/src/javascripts/projects/commercial/modules/hosted/youtube.js]: No, if we autoplay it manually (by clicking on the overlay). Yes if we will wait until the end of the movie to be auto-redirected automatically. In both situations, we track the autoplay event so it should be easy to detect why the video wasn't watched until the end.

**Pull request merged**